pbTests: Add support for ARM64 in QEMUPlaybookCheck

Willsparker opened the pull request:
Fixes: #1444  (This PR in conjunction with #1447 will fix 1444)

Add support for testing our playbooks on ARM64 Debian Buster through the `QEMUPlaybookCheck` Jenkins job. 
In addition to the changes done here, the `infra-softlayer-ubuntu1804-x64-1` machine has been updated to have the `ARM64.dsk.xz` file in `/qemu_base_images`. Like the other architectures, the disk image is 15GB, and has the user `linux` on it to run the playbook on.
The guide I used to setup a working Debian Buster ARM64 QEMU VM is from here https://wiki.debian.org/Arm64Qemu , using the `debian-10.4.3-20200610-openstack-arm64.qcow2` image found at https://cdimage.debian.org/cdimage/openstack/current/ .
I tested this on the `infra-softlayer-ubuntu1804-x64-1` machine and it ran through, though I can test it by configuring the `QEMUPlaybookCheck` job to pull from my git branch.

Willsparker commented:
Testing at : https://ci.adoptopenjdk.net/job/QEMUPlaybookCheck/44/ARCHITECTURE=arm64,label=vagrant/console 
It looks like it's managed to start it okay and the ansible output shows it's Debian 10 aarch64.
```
10:49:49 TASK [Debug : Display Debug Information] ***************************************
10:49:49 ok: [localhost] => {
10:49:49     "msg": [
10:49:49         "inventory_hostname: localhost ", 
10:49:49         "ansible_hostname: debian", 
10:49:49         "ansible_fqdn: debian", 
10:49:49         "ansible_user: linux", 
10:49:49         "ansible_default_ipv4.address: 10.0.2.15", 
10:49:49         "ansible_os_family: Debian ", 
10:49:49         "ansible_distribution: Debian ", 
10:49:49         "ansible_distribution_major_version: 10 ", 
10:49:49         "ansible_architecture: aarch64 ", 
```

sxa commented on ansible/pbTestScripts/qemu_test_script.sh:
```diff
@@ -139,24 +133,27 @@ done
 		"S390X" )
 			export MACHINE="s390-ccw-virtio";
 			export DRIVE="-drive file=$workFolder/S390X.dsk,if=none,id=hd0 -device virtio-blk-ccw,drive=hd0,id=virtio-disk0";
-			export COMMAND="s390x";;
```
Can we change the `COMMAND` variable to be `QEMUARCH` or something like that? Since it's not the full command name it's a bit misleading.

Willsparker replied:
Change made 👍

sxa commented on ansible/pbTestScripts/qemu_test_script.sh:
```diff
@@ -139,24 +133,27 @@ done
 		"S390X" )
 			export MACHINE="s390-ccw-virtio";
```
Not sure why we have `;` at the end of each line here - you only need `;;` at the end of the block

Willsparker replied:
I was probably missing `Java` or `C` that day ... Fixed now

sxa commented:
Looks like it's found a legitimate problem ;-)

sxa commented:
```

11:29:11 TASK [Common : Install GCC G++ on supported platforms] *************************
11:29:20 failed: [localhost] (item=g++-4.8) => {"ansible_loop_var": "item", "changed": false, "item": "g++-4.8", "msg": "No package matching 'g++-4.8' is available"}
11:29:29 failed: [localhost] (item=gcc-4.8) => {"ansible_loop_var": "item", "changed": false, "item": "gcc-4.8", "msg": "No package matching 'gcc-4.8' is available"}
```
Although it looks like it's excluded on Ubuntu 20 but not 18 based on https://github.com/AdoptOpenJDK/openjdk-infrastructure/blob/master/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Ubuntu.yml#L75

Willsparker commented:
As you've already seen (but putting it here for the benefit of others) , that's the `Ubuntu.yml` file, but the QEMUPlaybookCheck runs the `Debian.yml` file, but you've already put a fix in for it :-) #1454

Willsparker commented:
Ref: The false positive; `VPC` and `QPC` suffer from them. I'll address it in another PR :-) 

The checkout failed as it was looking for the `openj9` branch in `https://github.com/AdoptOpenJDK/openjdk-aarch64-jdk8u` which doesn't seem to exist. Do we build OpenJ9 for aarch64 or is it just Hotspot?

sxa commented:
We build both for aarch64, but OpenJ9 doesn't use the alternate aarch64-specific repository that this is picking up

Willsparker commented:
Which repository is it? I can't find `arm` or `aarch64` OpenJ9 for JDK8 - though I can for 11.
The script to start the build is `buildJDK.sh` which just calls `make-adopt-build-farm.sh` after setting variables, so it'll be an issue with that if it's setting the wrong repo.

sxa commented:
OpenJ9 aarch64 is only JDK11

Willsparker commented:
Ref: Latest commit;
- I've renamed `qemu_test_script.sh` to `qemuPlaybookCheck.sh` to match `vagrantPlaybookCheck.sh`.
- I've added the option to specify which jdk to build in `qemuPlaybookCheck.sh`.
- In `buildJDK.sh` I've restricted aarch64 from building JDK8/J9 - it defaults to JDK11/J9 now.

Willsparker commented:
I think you've jumped the[ gun slightly on that](https://ci.adoptopenjdk.net/job/QEMUPlaybookCheck/56/) - It's still trying to build JDK8/J9. My bad, I should've put the PR back into draft. I'll open another PR to get it working properly :-)
